best_practices.md

```diff
@@ -0,0 +1,195 @@
+format your code according to the chosen style.
+In addition to automatic formatters, tools for checking adherence to a style
+guide as well as for common programming mistakes are usually also available.
+These tools are called linters.
+It is recommended that you make use of an automatic formatter and linter.
+For example, a popular automatic formatter for Python is
+[black](https://black.readthedocs.io), while a good linter for Python is
+[pylint](https://pylint.org/).
+
+## Keep your code reliable and maintainable
+If you write large pieces of code, it is recommended that you implement some
+form of
+[automated testing](https://the-turing-way.netlify.app/reproducible-research/testing.html)
+to ensure that you do not loose more and more time testing that your code still
+produces the correct results as it grows.
+This can range from
+[unit tests](https://the-turing-way.netlify.app/reproducible-research/testing/testing-unittest.html),
+that test each function in your code individually, to
+[regression tests](https://the-turing-way.netlify.app/reproducible-research/testing/testing-acceptance-regression.html#regression-testing)
+that run larger portions of code and just check that the changes you make to your
+code do not change previously computed results that are not supposed to change.
+For most programming languages, tools are available to run all tests with a
+single command.
+A good choice for running Python tests is [pytest](https://docs.pytest.org).
+
+## Do code reviews
+If at some point you find yourself working on code with multiple people, it is
+highly recommended that you do
+[code reviews](https://the-turing-way.netlify.app/reproducible-research/reviewing.html)
+to improve the quality and reliability of your code.
+
+## Use online services
+
+A large number of services are available online to make developing and
+distributing software easier.
+There services are usually free for [open-source](#make-your-code-open-source)
+projects.
+
+### Host your code on a website with an issue tracker
+It is recommended that you develop your diagnostic script
+[in public](#open-source-with-a-license) on a modern code development platform
+like [GitHub](https://github.com/) or [GitLab](https://about.gitlab.com).
+This facilitates
+[collaboration](https://the-turing-way.netlify.app/collaboration/collaboration.html)
+and has the added benefit that you can make use
+of services for automatically hosting
+[documentation](#host-your-documentation-on-a-website),
+[automatic quality control](#use-a-code-quality-checker),
+[automatic testing](#automate-testing),
+and [automatic publication](#use-a-package-registry).
+
+Hold all conversations about developments of your code in the issue tracker,
+to make your work accessible to users and potential collaborators.
+If you have a conversation where decisions are made outside the issue tracker,
+make sure to summarize it in an issue.
+
+### Host your documentation on a website
```

Comment:
Add a section for writing the docs as well? With sphinx, [mkdocs](https://www.mkdocs.org/) ...

Reply:
Very good point, for some reason this is missing from The Turing Way, so I missed it